README.md

## Usage

You can change and setup your high designed CountNumberButton: 
 - boxheight and boxwidth
 - initial value
 - the minimum and maximum value
 - icon colors and size for each icon for left and right
 - onchanged you can use the final value for whatever you want to do with..
 - you can also choose your fav textcolor
 
## Example 

```dart
import 'package:flutter/material.dart';
import 'package:countnumberbutton/countnumberbutton.dart';

class ExampleWidget extends StatefulWidget {
  const ExampleWidget({Key? key}) : super(key: key);

  @override
  State<ExampleWidget> createState() => _ExampleWidgetState();
}

class _ExampleWidgetState extends State<ExampleWidget> {
  int newValue = 0;
  
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Container(
        decoration: BoxDecoration(
          border: Border.all(color: Colors.black)
        ),
        child: CountNumberButton(
          width: 250,
          height: 60,
          initValue: newValue,
          minValue: 0,
          maxValue: 10,
          iconColor: Colors.white,
          iconSize: 25.0,
          buttonColor: Colors.green,
          textColor: Colors.black,
          icon_left: Icon(Icons.remove),
          icon_right: Icon(Icons.add),
          onChanged: (value) {
            newValue = value;
          },
        )
      )
    );
  }
}
//if you use an initValue only as undeclared integer -> it'll have no relation and does not work like it have to do..
// FOLLOW THESE STEPS: 
//1. Create a Value like newValue as int like on the top of code newValue = 0 ..- init it in CountNumberButton Widget.. 
//2. - use onChanged method and set newValue as value to save the state of val to work with...
// NOW YOU HAVE YOUR COUNTNUMBERBUTTON .. PLEASE LIKE IT AND HAVE FUN WITH IT.. :D

```


CHANGELOG.md
## 0.0.6

* Configure your CountNumberButton. Resizable Counter Button. Changeable Background Colors & Icons.


test/counterbutton_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:counterbutton/counterbutton.dart';

void main() {}


lib/counterbutton.dart
import 'package:flutter/material.dart';

class CounterButton extends StatefulWidget {
  final double width;
  final double height;
  late int initValue;
  final int minValue;
  final int maxValue;
  final Color iconColor;
  final Color buttonColor;
  final Color textColor;
  final double iconSize;
  final Icon icon_left;
  final Icon icon_right;
  final ValueChanged onChanged;
  CounterButton({
    Key? key,
    required this.width,
    required this.height,
    required this.initValue,
    required this.minValue,
    required this.maxValue,
    required this.iconColor,
    required this.buttonColor,
    required this.textColor,
    required this.iconSize,
    required this.icon_left,
    required this.icon_right,
    required this.onChanged,
  }) : super(key: key);

  @override
  State<CounterButton> createState() => _CounterButtonState();
}

class _CounterButtonState extends State<CounterButton> {
  @override
  Widget build(BuildContext context) {
    return Container(
        width: widget.width,
        height: widget.height,
        child: Center(
            child: Row(children: <Widget>[
          Padding(
            padding: EdgeInsets.all(3),
            child: Container(
              width: widget.width / 3.1,
              height: widget.height * 0.95,
              decoration: BoxDecoration(
                  color: widget.buttonColor,
                  borderRadius: BorderRadius.only(
                      topLeft: Radius.circular(15.0),
                      bottomLeft: Radius.circular(15.0))),
              child: IconButton(
                  onPressed: () {
                    if (widget.initValue <= widget.minValue) {
                      setState(() {
                        widget.initValue = widget.minValue;
                      });
                    } else {
                      setState(() {
                        widget.initValue -= 1;
                      });
                      widget.onChanged(widget.initValue);
                    }
                  },
                  icon: widget.icon_left,
                  iconSize: widget.iconSize,
                  color: widget.iconColor),
            ),
          ),
          Padding(
            padding: EdgeInsets.all(3),
            child: Container(
                width: widget.width / 4,
                height: widget.height * 0.95,
                decoration: BoxDecoration(
                  color: Colors.white,
                ),
                child: Center(
                  child: Text(
                    '${widget.initValue}',
                    style: TextStyle(
                      fontSize: 16,
                      color: widget.textColor,
                    ),
                  ),
                )),
          ),
          Padding(
            padding: EdgeInsets.all(3),
            child: Container(
              width: widget.width / 3.1,
              height: widget.height * 0.95,
              decoration: BoxDecoration(
                  color: widget.buttonColor,
                  borderRadius: BorderRadius.only(
                      topRight: Radius.circular(15.0),
                      bottomRight: Radius.circular(15.0))),
              child: IconButton(
                  onPressed: () {
                    if (widget.initValue >= widget.maxValue) {
                      setState(() {
                        widget.initValue = widget.maxValue;
                      });
                    } else {
                      setState(() {
                        widget.initValue += 1;
                        widget.onChanged(widget.initValue);
                      });
                    }
                  },
                  icon: widget.icon_right,
                  iconSize: widget.iconSize,
                  color: Colors.white),
            ),
          ),
        ])));
  }
}


lib/src/countnumberbutton.dart
import 'package:flutter/material.dart';

class CountNumberButton extends StatefulWidget {
  CountNumberButton({
    Key? key,
    required this.width,
    required this.height,
    required this.initValue,
    required this.minValue,
    required this.maxValue,
    required this.iconColor,
    required this.buttonColor,
    required this.textColor,
    required this.iconSize,
    required this.icon_left,
    required this.icon_right,
    required this.onChanged,
  }) : super(key: key);

  final double width;
  final double height;
  int initValue;
  final int minValue;
  final int maxValue;
  final Color iconColor;
  final Color buttonColor;
  final Color textColor;
  final double iconSize;
  final Icon icon_left;
  final Icon icon_right;
  final ValueChanged onChanged;

  @override
  State<CountNumberButton> createState() => _CountNumberButtonState();
}

class _CountNumberButtonState extends State<CountNumberButton> {
  @override
  Widget build(BuildContext context) {
    return Container(
        width: widget.width,
        height: widget.height,
        child: Center(
            child: Row(children: <Widget>[
          Padding(
            padding: EdgeInsets.all(3),
            child: Container(
              width: widget.width / 3.1,
              height: widget.height * 0.95,
              decoration: BoxDecoration(
                  color: widget.buttonColor,
                  borderRadius: BorderRadius.only(
                      topLeft: Radius.circular(15.0),
                      bottomLeft: Radius.circular(15.0))),
              child: IconButton(
                  onPressed: () {
                    if (widget.initValue <= widget.minValue) {
                      setState(() {
                        widget.initValue = widget.minValue;
                      });
                    } else {
                      setState(() {
                        widget.initValue -= 1;
                      });
                      widget.onChanged(widget.initValue);
                    }
                  },
                  icon: widget.icon_left,
                  iconSize: widget.iconSize,
                  color: widget.iconColor),
            ),
          ),
          Padding(
            padding: EdgeInsets.all(3),
            child: Container(
                width: widget.width / 4,
                height: widget.height * 0.95,
                decoration: BoxDecoration(
                  color: Colors.white,
                ),
                child: Center(
                  child: Text(
                    '${widget.initValue}',
                    style: TextStyle(
                      fontSize: 16,
                      color: widget.textColor,
                    ),
                  ),
                )),
          ),
          Padding(
            padding: EdgeInsets.all(3),
            child: Container(
              width: widget.width / 3.1,
              height: widget.height * 0.95,
              decoration: BoxDecoration(
                  color: widget.buttonColor,
                  borderRadius: BorderRadius.only(
                      topRight: Radius.circular(15.0),
                      bottomRight: Radius.circular(15.0))),
              child: IconButton(
                  onPressed: () {
                    if (widget.initValue >= widget.maxValue) {
                      setState(() {
                        widget.initValue = widget.maxValue;
                      });
                    } else {
                      setState(() {
                        widget.initValue += 1;
                        widget.onChanged(widget.initValue);
                      });
                    }
                  },
                  icon: widget.icon_right,
                  iconSize: widget.iconSize,
                  color: Colors.white),
            ),
          ),
        ])));
  }
}
